app/models/manageiq/providers/amazon/storage_manager/s3/cloud_object_store_object.rb

```diff
@@ -18,6 +18,10 @@ def provider_object(connection = nil)
   end
 
   def raw_delete
-    with_provider_object(&:delete)
+    if key.end_with? "/" # delete object with subobjects (aka. folder)
```

Comment:
Thank you for opening this debase @gberginc. Please see my answer below - bottomline is that I would still merge this PR.

Reading the user documentation here [Listing Keys Hierarchically Using a Prefix and Delimiter](http://docs.aws.amazon.com/AmazonS3/latest/dev/ListingKeysHierarchy.html) I can agree to what you've found out @gberginc, objects are generally not in parent-children relation. On the other hand I also agree to Matouš who has reported the bug on Bugzilla: if S3 console mimics the parent-children behavior assuming `/` separator, why would we do it any different in MiQ?

But then based on [Working with Folders](http://docs.aws.amazon.com/AmazonS3/latest/UG/FolderOperations.html) I woud say that it is perfectly safe to merge this PR. The link says that object whose name ends with `/` actually **is** different from other objects:
```
An object named with a trailing "/" displays as a folder in the Amazon S3 console.
The Amazon S3 console does not display the content and metadata for such an object.
When copying an object named with a trailing "/" by using the Amazon S3 console, a
new folder is created in the destination location but the object's data and metadata
are not copied. 
```

@bronaghs, @Ladas may I ask for your opinion?